### PR TITLE
Feature(be) GRID-9112: Documentation for Credentials & Private S3 Datastore Support

### PR DESCRIPTION
**Pull request by rlizzo**

### CODE REVIEW

File: docs/features/datastores/2_Using Datastores/2_creating-datastores.md [hunk 1]
```diff
@@ -131,6 +131,20 @@ If you are using this flag via the Grid public cloud, then the source bucket sho
...
+### Creating Datastore From Private AWS S3 Buckets
+
+Grid now supports the ability to create datastores from private AWS s3 buckets when using
+the `--no-copy` mode via the CLI. In order to allow Grid to access your private buckets
+you should create an authorized AWS Role using the `grid credential create --type s3`
+command (explained in detail below), after which you can run the
+`grid datastore create S3://<private-bucket-name-here> --no-copy` command as usual (no
+modifications needed). If any of your registered `s3` credentials can access the s3 bucket
+path specified, then Grid will automatically use them when creating the datastore (and
+when using that datastore in a `run` or `session`)
```
[the remark this answers is not in the document]
Comment (RobertLaurella) [reply docs/features/datastores/2_Using Datastores/2_creating-datastores.md:144]: Is this option only available via the CLI and not the web UI? If so we need to flag to user

File: docs/platform/3_credentials.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+## Creating an `s3` Credential to Create Datastores from private AWS S3 Buckets 
+
+Grid now supports the ability to create datastores from private AWS s3 buckets when using
+the [`--no-copy` mode via the CLI](../features/datastores/2_Using%20Datastores/2_creating-datastores.md). In order to allow Grid to access your private buckets
+you should create an authorized AWS Role using the `grid credential create --type s3`
+command (explained in detail below), after which you can run the `grid datastore create
+S3://<private-bucket-name-here> --no-copy` command as usual (no modifications needed). If
+any of your registered `s3` credentials can access the s3 bucket path specified, then Grid
+will automatically use them when creating the datastore (and when using that datastore in
+a `run` or `session`)
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:38]:
```suggestion
Grid now supports the ability to create Datastores from private AWS s3 buckets by using
the `--no-copy` mode via the CLI. In order to allow Grid to access your private buckets,
yo'll need to create an authorized AWS Role using the `grid credential create --type s3`
command (explained in detail below). After, you can run the
`grid datastore create S3://<private-bucket-name-here> --no-copy` command as usual - no
modifications needed. If any of your registered `s3` credentials can access the s3 bucket
path specified, then Grid will automatically use them when creating the Datastore (and
when using that Datastore in a `run` or `session`)
```

File: docs/platform/3_credentials.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+## Access to Secrets (Teams Users)
+
+Credentials can be created by any user of the Grid platform (there is a limit of 50
+Credentials per user account). Only the user who created a credential (the owner) can
+delete the Credential. Any team-members of the User are able to utilize the credential in
+their own workflow (for example, while creating a datastore from a private s3 bucket)
+
+Sensitive credential type values can never be retrieved from Grid via the CLI. However,
+the values of non-private credential types (such as `--type s3`) can be viewed by the user
+who created the credential and any member of the User's team (for our `teams` tier users.)
```
[the remark this answers is not in the document]
Comment (RobertLaurella) [reply docs/platform/3_credentials.md:21]: Why is User capitalized in in lines 16 & 21?

File: docs/platform/3_credentials.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+# Credential Types
+
+At the moment the following credential types are supported:
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:25]:
```suggestion
At the moment, the following credential types are supported:
```

File: docs/platform/3_credentials.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+In order to provide Grid access to a private S3 bucket, you must configure an AWS IAM Role
+which is configured with an appropriate role trust policy and an attached permission
+policy which allows appropriate access. The trust and permission policy required are the
+minimal permission sets which Grid requires in order to list files in some s3 bucket
+prefix which you define and to "get" the objects from cloud storage. The `grid credential
+create` command will dynamically generate the IAM Role Trust Policy based on the cluster
+you are creating the datastore on. While all users will need to authorize the AWS account
+ID which the Grid Platform controlplane runs on, [bring-your-own-cloud (`BYOC`)
+users](./2_Custom%20Cloud%20Credentials/1_byoc.md) will also need to include the AWS
+account ID which the `BYOC` cluster runs on.
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:49]:
```suggestion
In order to provide Grid access to a private S3 bucket, you must first set up an AWS IAM Role configured with the appropriate permission policy. This is the absolute minimum permission Grid requires in order to list and retrieve files from an s3 bucket of your choice. The `grid credential create` command will dynamically generate the IAM Role Trust Policy based on the cluster on which you are creating the Datastore. While all users will need to authorize the AWS account ID on which the Grid Platform controlplane runs, [bring-your-own-cloud (`BYOC`) users](./2_Custom%20Cloud%20Credentials/1_byoc.md) will also need to include the AWS account ID on which the `BYOC` cluster runs. 
```

File: docs/platform/3_credentials.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+Please refer to the AWS documentation on [IAM Role Trust
+Policies](https://aws.amazon.com/blogs/security/how-to-use-trust-policies-with-iam-roles/)
+and [IAM Role Permission
+Policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) for more
+detailed information.
+
+:::
+
+We will illustrate the process of registering an IAM Role with Grid with the following
+example:
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:62]:
```suggestion
We will illustrate the process of registering an IAM Role with Grid using the following
example:
```

File: docs/platform/3_credentials.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+> I have an S3 bucket in an AWS account which contains private (sensitive) data of some
+> kind which I would like to use from a Grid Datastore. The bucket's name
+> `gridai-demo-bucket` and I would like to be able to create a datastore from any file
+> prefix (folder) in the bucket.
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:67]:
```suggestion
> I have an S3 bucket in an AWS account which contains private (sensitive) data that I would like to use from a Grid Datastore. The bucket's name is
> `gridai-demo-bucket` and I would like to create a Datastore from any file
> prefix (folder) in the bucket. 
```

Comment (RobertLaurella) [reply docs/platform/3_credentials.md:67]: is 'use from a Grid Datastore' correct here?

File: docs/platform/3_credentials.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+> I have an S3 bucket in an AWS account which contains private (sensitive) data of some
+> kind which I would like to use from a Grid Datastore. The bucket's name
+> `gridai-demo-bucket` and I would like to be able to create a datastore from any file
+> prefix (folder) in the bucket. 
+
+![](../../static/images/credentials/demo_bucket_contents.png)
+
+First: run the `grid credentials create --type s3` command in order to find the trust and
+permission policy which is needed.
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:72]:
```suggestion
First: run the `grid credentials create --type s3` command in order to find the needed trust and
permission policy.
```

File: docs/platform/3_credentials.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+name you wish to grant access to. More information can be found on the docs.
+
+Please Note: when creating the role name in the AWS console, the role name MUST 
+begin with the prefix: grid-s3-access- any valid characters can follow the 
+prefix.
+
+When complete, please enter the role ARN:
+```
+
+As you can see the command outputs the required trust and permission policies.
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:127]:
```suggestion
The command will output the required trust and permission policies. 
```

File: docs/platform/3_credentials.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+click "Create Role"
+
+![](../../static/images/credentials/role-console.png)
+
+I will select the "custom trust policy" setting, and copy/paste the trust policy output
+from the CLI into the `json` editor:
+
+![](../../static/images/credentials/select_trusted_entity.png)
+
+When prompted with the "add permissions to role" tab, I elect to create a new permission
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:140]:
```suggestion
When prompted with the "add permissions to role" tab, I create a new permission
```

File: docs/platform/3_credentials.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+policy by clicking on the "Create Policy" button
+
+![](../../static/images/credentials/add_permissions_to_role.png)
+
+A new tab will open (keep the old tab open as well), and I select the `json` tab, and
+copy/paste the permission policy into the editor. I also replace the text
+`<replace-with-bucket-name>` with by private bucket name I want to provide access to
+(i.e. `gridai-demo-bucket`).
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:148]:
```suggestion
`<replace-with-bucket-name>` with the name of the private bucket I want to provide access to
(i.e. `gridai-demo-bucket`). 
```

File: docs/platform/3_credentials.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+![](../../static/images/credentials/add_permissions_to_role.png)
+
+A new tab will open (keep the old tab open as well), and I select the `json` tab, and
+copy/paste the permission policy into the editor. I also replace the text
+`<replace-with-bucket-name>` with by private bucket name I want to provide access to
+(i.e. `gridai-demo-bucket`). 
+
+![](../../static/images/credentials/aws_permission_policy.png)
+
+I click next until I hit the "permission policy naming" tab, and I enter an appropriate
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:152]:
```suggestion
I click next until I reach the "permission policy naming" tab, and I enter an appropriate
```

File: docs/platform/3_credentials.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+![](../../static/images/credentials/permission_policy_naming.png)
+
+I click "create policy" and am presented with a message saying the operation was
+successful.
+
+![](../../static/images/credentials/permission_policy_created.png)
+
+At this point I go back to the old tab for attaching a permission policy to the Role which
+I had been at before. I "refresh" the list of policies by clicking the refresh symbol, and
+then select the `my-permission-policy-for-grid` policy name and hit next.
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:164]:
```suggestion
At this point, I go back to the previous tab for attaching a permission policy to the Role which
I had been at before. I "refresh" the list of policies by clicking the refresh symbol, and
then select the `my-permission-policy-for-grid` policy name and click next.
```

File: docs/platform/3_credentials.md [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+prefix: `grid-s3-access-`**. In this case I name the role
+`grid-s3-access-for-my-demo-bucket` and click create.
+
+![](../../static/images/credentials/role_naming.png)
+
+I am then presented with a success screen! The role has been created. 
+
+![](../../static/images/credentials/role_created.png)
+
+At this point I select the role, and click on the icon next to the "role ARN" in order to
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:179]:
```suggestion
I select the role and click on the icon next to the "role ARN" in order to
```

**Comment by essiequoi** [comment]
hey @rlizzo can you change base branch to dev?

**Comment by essiequoi** [comment]
going to make the dev branch the default to prevent this from happening again :)

File: docs/platform/3_credentials.md [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+When complete, please enter the role ARN: arn:aws:iam::880213070634:role/grid-s3-access-for-my-demo-bucket
+
+Success!
+```
+
+The role has been successfully added, and I can see it in the list when I run `grid
+credentials` or `grid credentials list`. 
+
+At this point I can run the regular commands to create a datastore and it will
+automatically pick the correct credential and create the datastore!
```
Comment (RobertLaurella) [inline docs/platform/3_credentials.md:200]:
```suggestion
automatically pick the correct credential and create the Datastore from my private s3 bucket!
```

File: docs/features/datastores/2_Using Datastores/2_creating-datastores.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rlizzo) [reply docs/features/datastores/2_Using Datastores/2_creating-datastores.md:144]: yes it is. will flag this in the credentials doc page.

File: docs/platform/3_credentials.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rlizzo) [reply docs/platform/3_credentials.md:21]: absolutely no reason.

**Comment by rlizzo** [comment]
TODO: rebuild click autodocs from prod CLI or revert changes

File: docs/features/datastores/2_Using Datastores/2_creating-datastores.md [hunk 16]
```diff
@@ -134,14 +134,14 @@ If you are using this flag via the Grid public cloud, then the source bucket sho
...
-you should create an authorized AWS Role using the `grid credential create --type s3`
-command (explained in detail below), after which you can run the
-`grid datastore create S3://<private-bucket-name-here> --no-copy` command as usual (no
-modifications needed). If any of your registered `s3` credentials can access the s3 bucket
-path specified, then Grid will automatically use them when creating the datastore (and
-when using that datastore in a `run` or `session`)
+Grid now supports the ability to create Datastores from private AWS s3 buckets by using
+the `--no-copy` mode via the CLI. In order to allow Grid to access your private buckets,
+you'll need to create an authorized AWS Role using the `grid credential create --type s3`
+command (explained in detail below). After, you can run the
```
Comment (essiequoi) [inline docs/features/datastores/2_Using Datastores/2_creating-datastores.md:140]:
```suggestion
command (explained in detail below). After creating a role, you can run the
```

File: docs/features/datastores/2_Using Datastores/2_creating-datastores.md [hunk 17]
```diff
@@ -134,14 +134,14 @@ If you are using this flag via the Grid public cloud, then the source bucket sho
...
+`grid datastore create S3://<private-bucket-name-here> --no-copy` command as usual - no
+modifications needed. If any of your registered `s3` credentials can access the s3 bucket
+path specified, then Grid will automatically use them when creating the Datastore (and
+when using that Datastore in a `run` or `session`)
```
Comment (essiequoi) [inline docs/features/datastores/2_Using Datastores/2_creating-datastores.md:144]:
```suggestion
when using that Datastore in a `run` or `session`).
```

File: docs/platform/3_credentials.md [hunk 18]
```diff
@@ -5,48 +5,39 @@ sidebar_label: Credentials
...
 
 ## Access to Secrets (Teams Users)
 
-Credentials can be created by any user of the Grid platform (there is a limit of 50
-Credentials per user account). Only the user who created a credential (the owner) can
-delete the Credential. Any team-members of the User are able to utilize the credential in
-their own workflow (for example, while creating a datastore from a private s3 bucket)
+Credentials can be created by any Grid user, with a limit of 50
+Credentials per account. Only the user who created a credential (the owner) can
+delete that Credential. All of the User's team members are able to utilize the credential in
```
Comment (essiequoi) [inline docs/platform/3_credentials.md:16]:
```suggestion
delete that credential. All of the user's team members are able to utilize the credential in
```

File: docs/platform/3_credentials.md [hunk 19]
```diff
@@ -5,48 +5,39 @@ sidebar_label: Credentials
...
+their own workflow (for example, while creating a Datastore from a private s3 bucket)
 
 Sensitive credential type values can never be retrieved from Grid via the CLI. However,
 the values of non-private credential types (such as `--type s3`) can be viewed by the user
 who created the credential and any member of the User's team (for our `teams` tier users.)
```
Comment (essiequoi) [inline docs/platform/3_credentials.md:21]:
```suggestion
who created the credential and any member of the user's team (for our `teams` tier users.)
```

File: docs/platform/3_credentials.md [hunk 20]
```diff
@@ -5,48 +5,39 @@ sidebar_label: Credentials
...
 
 # Credential Types
 
-At the moment the following credential types are supported:
+At the moment, the following credential types are supported:
 
 - `s3` (access for datastore creation from a private bucket)
```
Comment (essiequoi) [inline docs/platform/3_credentials.md:27]:
```suggestion
- `s3` (access for datastore creation from a private s3 bucket)
```

File: docs/platform/3_credentials.md [hunk 21]
```diff
@@ -5,48 +5,39 @@ sidebar_label: Credentials
...
-prefix which you define and to "get" the objects from cloud storage. The `grid credential
-create` command will dynamically generate the IAM Role Trust Policy based on the cluster
-you are creating the datastore on. While all users will need to authorize the AWS account
-ID which the Grid Platform controlplane runs on, [bring-your-own-cloud (`BYOC`)
-users](./2_Custom%20Cloud%20Credentials/1_byoc.md) will also need to include the AWS
-account ID which the `BYOC` cluster runs on. 
+Grid now supports the ability to create Datastores from private AWS s3 buckets by using
+the `--no-copy` mode via the CLI. In order to allow Grid to access your private buckets,
+you'll need to create an authorized AWS Role using the `grid credential create --type s3`
+command (explained in detail below). After, you can run the
```
Comment (essiequoi) [inline docs/platform/3_credentials.md:34]:
```suggestion
command (explained in detail below). After creating a role, you can run the
```

**Comment by essiequoi** [comment]
my biggest feedback here is for the credentials workflow. i think it needs to be organized into a series of steps, so that someone new to the process is able to quickly understand all of the steps that will be required to create a credential. right now it appears as a wall of text

**Comment by essiequoi** [comment]
@RobertLaurella can you formally approve if this one is ready?
I'll be merging this one as well as FSX docs next Tuesday

**Review by RobertLaurella — APPROVED** [review]

File: docs/platform/3_credentials.md [hunk 22]
```diff
@@ -0,0 +1,216 @@
...
+the values of non-sensitive credential types (such as `--type s3`) can be viewed by the user
+who created the credential and any member of the user's team (for our `teams` tier users.)
+
+### Credential Types
+
+At the moment, the following credential types are supported:
+
+- `s3` (access for datastore creation from a private s3 bucket)
+
+## How to Create an `s3` Credential for Private S3 Access
```
Comment (essiequoi) [inline docs/platform/3_credentials.md:29]:
```suggestion
## How to Create an `s3` Credential for Private S3 Access (BYOC users only)
```

**Pull request merged**